### ISSUE TITLE
MulticlassAUROC: Implement a multi-class version of the AUROC metric

### ISSUE THREAD

**Issue by NumesSanguis** [PyTorchLightning/pytorch-lightning#3304]
## 🚀 Feature

Create the metric MulticlassAUROC to allow for the AUROC metric to be used in multi-class problem settings. Or,
Expand the AUROC metric to support multi-class data, which would also directly solve this AUROC bug that instead gives a random value when used in multi-class problems: https://github.com/PyTorchLightning/pytorch-lightning/issues/3303

### Motivation

AUROC is a useful metric for model performance, but the current AUROC metric is not made for multi-class problems.
An out of the box working multi-class AUROC metric would be great.

### Pitch

According to [this post on the PyTorch Lightning forum](https://forums.pytorchlightning.ai/t/pytorch-lightning-auroc-value-for-multi-class-seems-to-be-completely-off-compared-to-sklearn-using-it-wrong/61/7), implementation would just require wrapping existing functionality together.

Implementing this metric would lower the barrier of its usage, instead of needing to be aware of the different metrics available that could be combined to effectively get his metric.

### Alternatives

- Use sklearn's implementation of AUROC: [`sklearn.metrics.roc_auc_score`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html).
This slows down the training, however, due to the need of transferring prediction and target data to CPU.
- Implement manually by combining [multiclass auc](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/functional/classification.py#L825) and a [multiclass roc calculation](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/functional/classification.py#L620)

### Additional context

- Relevant discussion: https://forums.pytorchlightning.ai/t/pytorch-lightning-auroc-value-for-multi-class-seems-to-be-completely-off-compared-to-sklearn-using-it-wrong/61/6
- Relevant AUROC bug when used for multi-class: https://github.com/PyTorchLightning/pytorch-lightning/issues/3303

**Comment by SkafteNicki** [comment]
It would be a great addition. @NumesSanguis would you be up for creating a PR that implements a MulticlassAUROC metric?

**Comment by ddrevicky** [comment]
Assign this to me please.
